llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -936,13 +937,83 @@ Error MCCASBuilder::createStringSection(
   return Error::success();
 }
 
+/// Reads and returns the length field of a debug info header contained in
+/// Reader, assuming Reader is positioned at the beginning of the section.
+/// The Reader's state is advanced to the first byte after the section.
+Expected<size_t> getSizeFromDebugHeaderAndSkip(BinaryStreamReader &Reader) {
```

Comment:
Maybe call this getSizeFromDebug*Info*HeaderAndSkip() since there will be many more similarly named sections that we need to handle?

Reply:
Well, I think this is fine @adrian-prantl, the DWARF headers follow the same format, so this is a generic function to get the length of any DWARF header

Reply:
I see: Maybe getSizeFromDwarfHeaderAndSkip() then?

Reply:
Will change to Adrian's suggestion